### PR TITLE
Bookstore: connect to API

### PR DESCRIPTION
**Pull request by ChaabaneAnas**
## CHANGES

- Fetch book list from Bookstore API on Books page load.

  > Add an Action Creator, that returns a function.
  > Fetch the list of books.
  > Dispatch another action (add an action type) that will pass API response to the reducer and update the state

- Refactor  add book and remove book features to persist  changes in the server

   >  Change existing Action Creators, so they return functions.
   >   Make changes by using API.
   >   Dispatch an action (action type that existed before refactor) that will update the state accordingly.

### CODE REVIEW

File: src/redux/Books/books.js [hunk 1]
```diff
@@ -1,36 +1,84 @@
...
+    },
+    body: JSON.stringify({
+      item_id: id,
+    }),
+  }).then(() => {
+    dispatch({
+      type: REMOVE,
+      payload: id,
+    });
+  }).catch((erorr) => console.log(erorr));
```
Comment (Davidosky007) [inline src/redux/Books/books.js]: * Kindly remove all console.log statement from your codebase as it isn't best practice also sensitive informations might be logged into the console

File: src/redux/Books/books.js [hunk 2]
```diff
@@ -1,36 +1,84 @@
...
+      Accept: 'application/json',
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify({
+      item_id: book.id,
+      title: book.title,
+      author: book.author,
+      category: 'Fiction',
+    }),
+  }).then(() => {
```
Comment (Davidosky007) [inline src/redux/Books/books.js]: * As stated in the [project requirements](https://github.com/microverseinc/curriculum-react-redux/blob/main/bookstore/connec_to_api.md#:~:text=Use%20fetch%20or%20axios%20with%20createAsyncThunk%20for%20making%20API%20calls.) Kindly use createAsyncThunk for your API calls

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
Hi @ChaabaneAnas 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there
## Highlights
* No linter errors 👍 
* Well written PR description 👍 
* Good looking readme file 👍 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @davidsoky007 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/Books/books.js (hunk 2 above)
Comment (ChaabaneAnas) [reply src/redux/Books/books.js]: @Davidosky007,
 i think it's only required for fetching the data and not posting it

**Review by phelian23 — APPROVED** [review]
Hi @ChaabaneAnas,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
✅ You used Gitflow correctly
✅ No linter error
✅ Good descriptive PR 👍
✅ Good commit messages 👍

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**